emu_mps/mps.py

```diff
@@ -307,6 +307,30 @@ def overlap(self, other: State, /) -> torch.Tensor:
         """
         return torch.abs(self.inner(other)) ** 2  # type: ignore[no-any-return]
 
+    def entanglement_entropy(self, mps_site: int) -> torch.Tensor:
+        """
+        Returns
+        the Von Neumann entanglement entropy of the state `mps` at the bond between sites b and b+1
+        S = -Σᵢsᵢ² log(sᵢ²)),
+        where sᵢ are the singular values at the chosen bond.
+        """
+        # orthogonalize the MPS at site b
+        self.orthogonalize(mps_site)
+
+        # perform svd on reshaped matrix at site b
+        matrix = self.factors[mps_site].flatten(end_dim=1)
+        s = torch.linalg.torch.linalg.svdvals(matrix)
+
+        # Calculate entropy from singular values 's'
+        s_sqrd = s**2
+        s_sqrd = s_sqrd[s_sqrd > 1e-12]
+
+        s_e = -torch.sum(s_sqrd * torch.log(s_sqrd))
```

Comment:
```suggestion
        s = torch.linalg.svdvals(matrix)
        s_e = torch.special.entr(s**2)
```
see [torch.special.entr](https://pytorch.org/docs/stable/special.html#torch.special.entr)

Reply:
so nice, I didn't know about the entropy function